core/src/main/scala/com/databricks/labs/remorph/RootTableIdentifier.scala

```diff
@@ -0,0 +1,79 @@
+package com.databricks.labs.remorph
+
+import com.databricks.labs.remorph.parsers.{intermediate => ir}
+import com.databricks.labs.remorph.transpilers.Transpiler
+
+private case class ChildActionMap(name: String, action: Action.Operation)
+
+class RootTableIdentifier(transpiler: Transpiler) {
```

Comment:
```suggestion
class RootTableIdentifier(transpiler: Transpiler, queries: Seq[String]) {
```
1. let's make this interface more robust and immutable by shifting queries to constructor arg
2. introduce `Parser` interface with `parse(String): LogicalPlan` as a single public method.

Reply:
We shouldn't move the queries to Contructor, we can have `queries`, `etl` or `workflows` 
I didn't understand the need parser interface.

Reply:
I have created a parser interface added them in `transpiler` let me know if you want me to implement parser separately.